BDMS 221 & 227: function to retrieve polymorphic table record

jacob-a-brown opened the pull request:
<!--
- Title the PR with the JIRA project & ticket number and short description (e.g., BDMS-123: Example display bug), or NO TICKET
- Keep sections short and scannable.
-->
### **Why**
_This PR addresses the following problem / context:_
- Records from the `parameter` and `status_history` tables need to be easily retrieved for schemas and tests

### **How**
_Implementation summary - the following was changed / added / removed:_
- A function to retrieve a record was put into `services/util.py`

### **Notes**
_Any special considerations, workarounds, or follow-up work to note?_
- This only retrieves one function in a sorted list by `start_date`. I don't think that we need one to retrieve _all_ records since that is easily done via the attribute from the mixins.

jacob-a-brown committed:
feat: get polymorphic record via function

jirhiker reviewed (commented):
where is this being used

jacob-a-brown commented:
I'll use it in tests. I initially thought I'd use it throughout, but it may not actually be needed (I'm still developing responses, so TBD). From [your comment on the well additional info PR](https://github.com/DataIntegrationGroup/NMSampleLocations/pull/237#discussion_r2505837705) I figured I'd put it in `services/util.py`, but if it's just used in tests should it go in `tests/__init__.py`?

jacob-a-brown committed:
refactor: latest record must have null end date

jacob-a-brown committed:
refactor: move polymorphic record retrival to tests

jacob-a-brown commented:
I updated this to make sure `end_date` is not null, but since `status_history` is eagerly loaded for each `thing` object I don't need to use `polymorphic_relationship` in the API. So I've moved it to `tests/__init__.py` for use in tests only.
